### PR TITLE
script with types

### PR DESCRIPTION
**Pull request by mynameisrufus**

### CODE REVIEW

File: src/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  type: 'object',
+  properties: {
+    port: { type: 'integer' },
+    apiEndpoint: { type: 'string' }
+  },
+  required: ['port', 'apiEndpoint'],
+  additionalProperties: false
+}
+
+config.load({
```
Comment (nguyenchr) [inline src/index.test.ts]: can assert that the config.get() values are as expected?

Comment (mynameisrufus) [reply src/index.test.ts]: that it errors?

Comment (nguyenchr) [reply src/index.test.ts]: why it errors? shouldn't it get the values from `js.js` and `ts.ts`?

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { load } from './load'
+import { get as config } from './config'
+
+export { load, config }
```
Comment (nguyenchr) [inline src/index.ts]: original code exported this as `get` not `config`

File: src/config.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+let config: LibConfigOptions | null = null
+
+export function get () {
+  if (!config) {
+    throw new Error('Config has not been loaded')
+  }
+  return config
+}
+
+export function set (_config: LibConfigOptions) {
```
Comment (nguyenchr) [inline src/config.ts]: I don't think that type is correct
this is the actual application config, not the options

Comment (mynameisrufus) [reply src/config.ts]: In my opinion your are completely correct

File: src/load.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+  if (!schema) {
+    throw new Error('Error loading config, you must provide a config schema')
+  }
+
+  console.log(`Loading config for env ${environment}`)
+  const configDirectory = path.resolve(configDir || 'config')
+  const configFile = path.resolve(path.join(configDirectory, environment))
+
+  if (fs.existsSync(`${configFile}.js`)) {
```
Comment (nguyenchr) [inline src/load.ts]: I think this if and the one below are the same except for different extensions, so maybe can DRY

File: src/types.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+export interface JSONSchema {
+  type: string
+  properties: {
+    [name: string]: {
```
Comment (nguyenchr) [inline src/types.ts]: I don't think this is right, properties can be super nested

File: test/config/ts.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export const config = {
+  port: 8080,
```
Comment (nguyenchr) [inline test/config/ts.ts]: can we change this to `port: 9090` 
then in index.test.ts we can assert that the js load = 8080 and the ts load = 9090
if this passes then this module has no bugs ever

**Pull request merged**